hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/schemas/V0570TssBaseSchema.java

```diff
@@ -60,6 +78,19 @@ public void migrate(@NonNull final MigrationContext ctx) {
         if (tssStatusState.get() == null) {
             tssStatusState.put(new TssStatus(WAITING_FOR_ENCRYPTION_KEYS, ACTIVE_ROSTER, Bytes.EMPTY));
         }
+
+        // remove TssEncryptionKeys from state if node ids are not present
+        // in both active and candidate roster's entries
+        final var tssStore = tssStoreFactory.apply(ctx.newStates());
+        tssStore.removeIfNotPresent(readableRosterStoreSupplier.get().getCombinedRosterEntriesNodeIds());
```

Comment:
Yeah, that makes sense, thank you!

Reply:
After thinking about this more I believe you're right, using a supplier is more explicit and doesn't depend on the implementation of the underlying states (filtered or unfiltered). 

That said, let's find a way to provide a supplier that doesn't require keeping a reference to a particular instance of state in the `Hedera` class (in particular, this code): 

```java
    /**
     * When applying and migrating schemas to a target state, it is set here to support
     * giving the {@link RosterService} schemas access to a {@link ReadablePlatformStateStore}
     * and {@link TssBaseService} schemas access to a {@link ReadableRosterStore}
     * before the roster lifecycle is adopted.
     */
    @Nullable
    @Deprecated
    private State initState;
```

Platform counts the number of references to a state before safely disposing of it. The similar case you mentioned, of roster accessing platform state, doesn't require keeping a reference to the initial state; can we emulate that?

Reply:
It does here in [Hedera.java](https://github.com/hashgraph/hedera-services/blob/52c9203d82b729d6259d412beb8a4b1fef739f75/hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java#L461-L464) when creating the `RosterService`

Reply:
@tinker-michaelj is it ok to keep a reference to a specific state like this? The intent is only to read from it during schema migrations, but IIRC platform won't be able to clean up this state because an active reference is kept. Seems like this would cause a memory leak.